File: consulti-coverage-test.js
#!/usr/bin/env node
/**
 * Consulti coverage test — 5 veins, real DB targets.
 * Burns ~5 verify + ~6 lead credits maximum.
 */
const KEY = process.env.CONSULTI_API_KEY;
if (!KEY) { console.error("CONSULTI_API_KEY not set"); process.exit(1); }

const BASE = "https://www.consulti.ai/api/v1";
const H = { "Authorization": `Bearer ${KEY}`, "Content-Type": "application/json" };

const sleep = ms => new Promise(r => setTimeout(r, ms));

async function post(path, body) {
  try {
    const r = await fetch(`${BASE}${path}`, { method: "POST", headers: H, body: JSON.stringify(body) });
    const text = await r.text();
    let json; try { json = JSON.parse(text); } catch { json = { raw: text }; }
    return { status: r.status, body: json };
  } catch (e) { return { status: 0, body: { error: e.message } }; }
}

async function get(path) {
  const r = await fetch(`${BASE}${path}`, { headers: H });
  return r.json();
}

function domainOf(url) {
  if (!url) return null;
  try {
    const u = url.includes("://") ? new URL(url) : new URL(`http://${url}`);
    let host = u.hostname.toLowerCase().replace(/^www\./, "");
    const social = ["facebook.com", "instagram.com", "google.com", "linkedin.com", "x.com", "twitter.com", "m.facebook.com"];
    if (social.some(s => host.includes(s))) return null;
    return host;
  } catch { return null; }
}

function summarise(label, res) {
  const { status, body } = res;
  if (status !== 200) {
    console.log(`  → HTTP ${status}: ${JSON.stringify(body).slice(0, 200)}`);
    return;
  }
  if (!body.success) {
    console.log(`  → FAIL: ${JSON.stringify(body).slice(0, 200)}`);
    return;
  }
  const d = body.data || {};
  if ("status" in d && "is_deliverable" in d) {
    console.log(`  → verify: status=${d.status} deliverable=${d.is_deliverable} catch_all=${d.is_catch_all} role=${d.is_role_account} cached=${d.cached} cost=${body.credits_used ?? d.credits_used}`);
  } else if (Object.keys(d).length === 0) {
    console.log(`  → NO MATCH (free) | cost=${body.credits_used ?? 0}`);
  } else {
    console.log(`  → MATCH: ${d.first_name} ${d.last_name} | ${d.job_title} @ ${d.company_name} | ${d.email} (${d.email_status}) | ${d.linkedin_url || '-'} | ${d.city || '-'}, ${d.country || '-'} | emp=${d.employee_count || '-'} | cost=${body.credits_used}`);
  }
}

(async () => {
  console.log("=".repeat(80));
  console.log("CREDITS BEFORE");
  console.log((await get("/credits")).data);
  console.log("=".repeat(80));

  // Test 1 — verify two known emails (info@ role + named alias)
  console.log("\n[TEST 1] /verify on existing KSD emails (does Consulti agree with our pipeline?)");
  for (const [label, email] of [
    ["Fiona Walker (info@ role)", "info@fionawalkerphotography.co.uk"],
    ["Mounting Stone (andrew@ named)", "andrew@themountingstone.co.uk"],
  ]) {
    console.log(` - ${label}: ${email}`);
    summarise(label, await post("/verify", { email }));
    await sleep(1500);
  }

  // Test 2 — find-by-name with no real domain (social-only "website")
  console.log("\n[TEST 2] /leads/find-by-name on UFH clubs with no real domain");
  for (const [club, raw] of [
    ["Osterley Rangers", "https://www.instagram.com/osterley.rangers/"],
    ["Hindsford AFC", "https://m.facebook.com/hindsford.tonics.7"],
  ]) {
    const d = domainOf(raw);
    console.log(` - ${club} (site=${raw}) → domain=${d}`);
    if (!d) { console.log("  → SKIPPED (no real domain — Consulti can't help)"); continue; }
    summarise(club, await post("/leads/find-by-name", { first_name: club.split(" ")[0], last_name: "Secretary", domain: d }));
    await sleep(1500);
  }

  // Test 3 — enrich on UK journalists (do they have media-people data?)
  console.log("\n[TEST 3] /leads/enrich on UFH journalists (UK media coverage check)");
  for (const [name, email] of [
    ["Andy Schooler", "andy@andyschoolermedia.com"],
    ["Helen Clarke", "helen.clarke@spektrix.com"],
  ]) {
    console.log(` - ${name}: ${email}`);
    summarise(name, await post("/leads/enrich", { email }));
    await sleep(1500);
  }

  // Test 4 — find-by-name on KSD owners w/ name+domain, no email (gap-fill)
  console.log("\n[TEST 4] /leads/find-by-name on KSD owners w/ name+domain, no email");
  for (const [first, last, site, biz] of [
    ["Paul", "Bygraves", "http://www.wearesapphire.co.uk/", "Sapphire (accountant)"],
    ["Derek", "Chadderton", "http://derekchadderton.co.uk/", "Actree Accountancy"],
  ]) {
    const d = domainOf(site);
    console.log(` - ${biz}: ${first} ${last} @ ${d}`);
    summarise(biz, await post("/leads/find-by-name", { first_name: first, last_name: last, domain: d }));
    await sleep(1500);
  }

  // Test 5 — verify on suspicious scraped rows
  console.log("\n[TEST 5] /verify on suspicious scraped rows (data hygiene)");
  for (const [label, email] of [
    ["Happycups sales@", "sales@happycups.co.uk"],
    ["Let Loose info@", "info@letlooseplay.co.uk"],
  ]) {
    console.log(` - ${label}: ${email}`);
    summarise(label, await post("/verify", { email }));
    await sleep(1500);
  }

  console.log("\n" + "=".repeat(80));
  console.log("CREDITS AFTER");
  console.log((await get("/credits")).data);
  console.log("=".repeat(80));
})();

File: consulti-coverage-test.py
#!/usr/bin/env python3
"""
Consulti coverage test — 5 veins, real DB targets.
Burns ~5 verify credits + ~6 lead credits max.
"""
import os, sys, json, time, urllib.parse
import requests

KEY = os.environ.get("CONSULTI_API_KEY")
if not KEY:
    print("ERROR: CONSULTI_API_KEY not set", file=sys.stderr); sys.exit(1)

BASE = "https://www.consulti.ai/api/v1"
H = {"Authorization": f"Bearer {KEY}", "Content-Type": "application/json"}

def post(path, body):
    try:
        r = requests.post(f"{BASE}{path}", headers=H, json=body, timeout=30)
        return r.status_code, (r.json() if r.headers.get("content-type","").startswith("application/json") else r.text)
    except Exception as e:
        return 0, {"error": str(e)}

def get(path):
    r = requests.get(f"{BASE}{path}", headers=H, timeout=15)
    return r.json()

def domain_of(url):
    if not url: return None
    try:
        p = urllib.parse.urlparse(url if "://" in url else f"http://{url}")
        host = (p.netloc or p.path).lower().split("/")[0]
        if host.startswith("www."): host = host[4:]
        # Skip social URLs — not real domains
        if any(s in host for s in ["facebook.com","instagram.com","google.com","linkedin.com","x.com","twitter.com"]):
            return None
        return host
    except: return None

def summarise(label, body):
    if not isinstance(body, dict):
        print(f"  → raw: {body[:200]}"); return
    if not body.get("success"):
        print(f"  → FAIL: {body.get('error') or body}"); return
    d = body.get("data") or {}
    if "status" in d:  # verify
        print(f"  → verify: status={d.get('status')} deliverable={d.get('is_deliverable')} catch_all={d.get('is_catch_all')} role={d.get('is_role_account')} cached={d.get('cached')} cost={body.get('credits_used', d.get('credits_used'))}")
    else:  # lead
        if d:
            print(f"  → MATCH: {d.get('first_name')} {d.get('last_name')} | {d.get('job_title')} @ {d.get('company_name')} | {d.get('email')} ({d.get('email_status')}) | {d.get('linkedin_url')} | {d.get('city')}, {d.get('country')} | emp={d.get('employee_count')} | cost={body.get('credits_used')}")
        else:
            print(f"  → NO MATCH (free) | cost={body.get('credits_used')}")

# --- before
print("=" * 80)
print("CREDITS BEFORE")
print(get("/credits")["data"])
print("=" * 80)

# Test 1 — verify two emails we already have
print("\n[TEST 1] /verify on existing KSD emails (does Consulti agree with our pipeline?)")
for label, email in [
    ("Fiona Walker (info@ role)", "info@fionawalkerphotography.co.uk"),
    ("Mounting Stone (andrew@ named)", "andrew@themountingstone.co.uk"),
]:
    print(f" - {label}: {email}")
    code, body = post("/verify", {"email": email}); summarise(label, body); time.sleep(1.5)

# Test 2 — find-by-name on a UFH club WITHOUT a proper domain (predicted miss)
print("\n[TEST 2] /leads/find-by-name on UFH clubs with no real domain (free miss prediction)")
for club_name, raw_site in [
    ("Osterley Rangers", "https://www.instagram.com/osterley.rangers/"),
    ("Hindsford AFC", "https://m.facebook.com/hindsford.tonics.7"),
]:
    d = domain_of(raw_site)
    print(f" - {club_name} (site={raw_site}) → domain={d}")
    if d:
        code, body = post("/leads/find-by-name", {"first_name": club_name.split()[0], "last_name": "Secretary", "domain": d})
        summarise(club_name, body)
    else:
        print("  → SKIPPED (no real domain — Consulti can't help here)")
    time.sleep(1.5)

# Test 3 — enrich on journalists (does Consulti know UK media people?)
print("\n[TEST 3] /leads/enrich on UFH journalists (UK media coverage check)")
for name, email in [
    ("Andy Schooler", "andy@andyschoolermedia.com"),
    ("Helen Clarke", "helen.clarke@spektrix.com"),
]:
    print(f" - {name}: {email}")
    code, body = post("/leads/enrich", {"email": email}); summarise(name, body); time.sleep(1.5)

# Test 4 — find-by-name on KSD businesses where we have owner first+last+domain but NO email
print("\n[TEST 4] /leads/find-by-name on KSD owners w/ name+domain, no email (gap-fill potential)")
for first, last, site, biz in [
    ("Paul", "Bygraves", "http://www.wearesapphire.co.uk/", "Sapphire (accountant)"),
    ("Derek", "Chadderton", "http://derekchadderton.co.uk/", "Actree Accountancy"),
]:
    d = domain_of(site)
    print(f" - {biz}: {first} {last} @ {d}")
    code, body = post("/leads/find-by-name", {"first_name": first, "last_name": last, "domain": d})
    summarise(biz, body); time.sleep(1.5)

# Test 5 — verify on a likely-bad scraped email (the "Happycups" garbage row)
print("\n[TEST 5] /verify on a known-suspicious scraped row (data hygiene test)")
for label, email in [
    ("Happycups sales@", "sales@happycups.co.uk"),
    ("Let Loose info@", "info@letlooseplay.co.uk"),
]:
    print(f" - {label}: {email}")
    code, body = post("/verify", {"email": email}); summarise(label, body); time.sleep(1.5)

print("\n" + "=" * 80)
print("CREDITS AFTER")
print(get("/credits")["data"])
print("=" * 80)
